File: Question 0199 - Binary Tree Right Side View/src/Solution.java
import java.util.*;
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        ArrayList<Integer> ans = new ArrayList<>();
        ans = solveRight(root, ans, 0);
        return ans;
    }
    ArrayList<Integer> solveRight(TreeNode root, ArrayList<Integer> ans, int level)
    {
        //base case
        if(root==null) return ans;

        //we entered into a new level
        if(level == ans.size()) ans.add(root.val);

        //recursive call
        ans= solveRight(root.right, ans, level+1);
        ans = solveRight(root.left, ans, level+1);

        return ans;
    }
}

File: Question 0199 - Binary Tree Right Side View/src/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    vector<int> rightSideView(TreeNode *root)
    {
        vector<int> ans;
        if (!root)
            return ans;

        queue<TreeNode *> q;
        q.push(root);

        while (!q.empty())
        {
            int sz = q.size();
            int right;

            for (int i = 0; i < sz; i++)
            {
                TreeNode *tmp = q.front();
                q.pop();

                right = tmp->val;

                if (tmp->left)
                    q.push(tmp->left);

                if (tmp->right)
                    q.push(tmp->right);
            }

            ans.push_back(right);
        }

        return ans;
    }
};